### ISSUE TITLE
Improving touch support and keyboard navigation

### ISSUE THREAD

**Issue by Garbee** [devtools-html/debugger.html#1768]
Currently the tools have useful functions hidden behind hover events (close/remove buttons in particular) and this does two things that hurt users:

1. Developers with touch devices have a harder time interacting with the tools. Whether this is just because they feel like it in a moment or because touch is more accessible to them then using a mouse.
2. It makes certain functions far more difficult (if not impossible) to reach using a keyboard. This is a major hindrance to users that need accessibility within the tools.

Check out this example using the breakpoints remove button with hover required to trigger:

![not-tabbable](https://cloud.githubusercontent.com/assets/868301/22151834/dda64dce-deed-11e6-9e88-704a22aa7d77.png)

The tab navigation goes to the *toggle* for the breakpoint, but the developer has no way to trigger removing it. The next viewport tab goes over to the file manager without any callstack or scopes to go through.

For the breakpoint remove button, we need to make the removal button constant and give it a good `aria-label` for screen readers. That should do the trick.

Not *everything* can be made touchable, like hover on variable tooltips when those land. However, for actions that developers ask the tools to carry out for them should be accessible by everyone.

**Comment by jasonLaster** [comment]
Thanks @Garbee.

I'm really glad that you're bringing up accessibility as I'd like for us to have some clear goals and then we can measure our progress and prevent regressions.


** touch accessibility** personally, I'd like to do well here but acknowledge that it might be hard to be 100%.
* we use context menus to hide a lot of extra functionality that we should audit as well
* *hover* i believe we only hide close buttons for the moment. The biggest use case here is removing breakpoints, which is still two taps away. Tap on the breakpoint in the list, tap on the breakpoint marker.
* if we're shooting for windows laptops, we *can* assume that the user could always still use the mouse for other actions like close a tab.

**aria labels & tooltips** I'd like to have every action clearly labeled for new users and screen readers. We should be able to use a tool that counts buttons with / without a label.

**semantic elements** I really like what you started doing with switching our div/spans to buttons, labels, etc. We should be able to flag spans/divs that have event handlers and make sure that we don't regress.

**keyboard navigation** `<tab>`, `<enter>`, `<space>`, `<arrows>` are really important to get right. We should have some integration tests that lock down this behavior and prevent regressions. we should have happy paths to all of the major functionality.

**focus event** if we're showing a button on hover, we should also show it on container focus and make it tabbable. We should have integration tests here too.

Lastly, this is ofcourse a lot of work, but I think it's important that we set up some standards. I'd love to see us create a tracking issue and come up with some good small next steps and get community help here.

CC @npsecoding @clarkbw

**Comment by jasonLaster** [comment]
I'm closing this issue, labeling it `low-priority`, and moving it the enhancements board.

I'd like to revisit this in the future and see how we're doing. I believe we're improving in these respects:

* our keyboard navigation, 
* aria labels and tooltips
* semantic elements
* focus / selected states
